icepyx/core/query.py

```diff
@@ -1,445 +1,56 @@
-    available ICESat-2 data products using temporal and spatial input parameters.
-    Allows the easy input and formatting of search parameters to match the
-    NASA NSIDC DAAC and (development goal-not yet implemented) conversion to multiple data types.
-    Expands the superclass GenQuery.
-
-    See the doc page for GenQuery for details on temporal and spatial input parameters.
-
-    Parameters
-    ----------
-    product : string
-        ICESat-2 data product ID, also known as "short name" (e.g. ATL03).
-        Available data products can be found at: https://nsidc.org/data/icesat-2/data-sets
-    version : string, default most recent version
-        Product version, given as a 3 digit string.
-        If no version is given, the current version is used. Example: "006"
-    cycles : string or a list of strings, default all available orbital cycles
-        Product cycle, given as a 2 digit string.
-        If no cycle is given, all available cycles are used. Example: "04"
-    tracks : string or a list of strings, default all available reference ground tracks (RGTs)
-        Product track, given as a 4 digit string.
-        If no track is given, all available reference ground tracks are used.
-        Example: "0594"
-    auth : earthaccess.auth.Auth, default None
-        An earthaccess authentication object. Available as an argument so an existing
-        earthaccess.auth.Auth object can be used for authentication. If not given, a new auth
-        object will be created whenever authentication is needed.
-
-    Returns
-    -------
-    query object
-
-    Examples
-    --------
-    Initializing Query with a bounding box.
-
-    >>> reg_a_bbox = [-55, 68, -48, 71]
-    >>> reg_a_dates = ['2019-02-20','2019-02-28']
-    >>> reg_a = Query('ATL06', reg_a_bbox, reg_a_dates)
-    >>> print(reg_a)
-    Product ATL06 v006
-    ('bounding_box', [-55.0, 68.0, -48.0, 71.0])
-    Date range ['2019-02-20', '2019-02-28']
-
-    Initializing Query with a list of polygon vertex coordinate pairs.
-
-    >>> reg_a_poly = [(-55, 68), (-55, 71), (-48, 71), (-48, 68), (-55, 68)]
-    >>> reg_a_dates = ['2019-02-20','2019-02-28']
-    >>> reg_a = Query('ATL06', reg_a_poly, reg_a_dates)
-    >>> reg_a.spatial_extent
-    ('polygon', [-55.0, 68.0, -55.0, 71.0, -48.0, 71.0, -48.0, 68.0, -55.0, 68.0])
-
-    Initializing Query with a geospatial polygon file.
-
-    >>> from pathlib import Path
-    >>> aoi = Path('./doc/source/example_notebooks/supporting_files/simple_test_poly.gpkg').resolve()
-    >>> reg_a_dates = ['2019-02-22','2019-02-28']
-    >>> reg_a = Query('ATL06', str(aoi), reg_a_dates)
-    >>> print(reg_a)
-    Product ATL06 v006
-    ('polygon', [-55.0, 68.0, -55.0, 71.0, -48.0, 71.0, -48.0, 68.0, -55.0, 68.0])
-    Date range ['2019-02-22', '2019-02-28']
-
-    See Also
-    --------
-    GenQuery
-    """
 
+class Query(BaseQuery):
+    _temporal: Union[tp.Temporal, None]
```

Comment:
As I went to update the API docs for this module, I realized I'm still struggling to see how BaseQuery fits into the bigger picture. My current understanding is that it was a useful distinction while we were working on migrating functions to highlight which ones needed to be touched/updated, but is no longer a useful division of functions. Fundamentally, nothing was replaced by "direct" earthaccess calls (e.g. the user still needs to create a `Query` object), and if anything the things in query.py should ultimately be the IS2 specific stuff that builds on the earthaccess GenQuery/BaseQuery-type object that would be created by earthaccess, not icepyx (thinking ahead to the plugin architecture). Thoughts?